### PR TITLE
Support other repos using the multidim interop tests

### PR DESCRIPTION
**Pull request by MarcoPolo**
Features:
- Filter tests with a flag
- Load container image from runtime param.
- Add extra versions with a runtime parameter. This allows other repos to define their "head" version to test against.
- Add emit only option, which always users to emit the compose spec that they can then manually run.

How to use this with other repos:
1. In a job, build a container image. [Example](https://github.com/libp2p/go-libp2p/pull/1991/files#diff-f6e386535c20db77aa925c6a8ed443534c486443f73ca333f77ef75493698128R15)
2. Upload the container image as an artifact (this is the only way to pass data between jobs in GH Actions). [Example](https://github.com/libp2p/go-libp2p/pull/1991/files#diff-f6e386535c20db77aa925c6a8ed443534c486443f73ca333f77ef75493698128R23)
3. Run this workflow from the other repo as another job that depends on (i.e. `needs`) the previous job. And pass in the proper input parameters. [Example](https://github.com/libp2p/go-libp2p/pull/1991/files#diff-f6e386535c20db77aa925c6a8ed443534c486443f73ca333f77ef75493698128R28)

Example run that tests across versions and impl for go-libp2p-head: https://github.com/libp2p/go-libp2p/actions/runs/3894748269

### CODE REVIEW

**Comment by MarcoPolo** [comment]
tagging either @mxinden or @marten-seemann for review please :)

File: .github/actions/setup-testground/action.yml [hunk 1]
```diff
@@ -1,82 +0,0 @@
-name: start testground
```
Comment (marten-seemann) [inline .github/actions/setup-testground/action.yml:1]: 🎉

File: .github/workflows/run-testplans.yml [hunk 2]
```diff
@@ -7,6 +7,24 @@ on:
...
         required: true
         type: string
+      extra-versions:
+        description: artifact name for the extra-versions.json file
+        required: false
+        type: string
+      image-tar:
+        description: artifact name for the image.tar(s) to import
+        required: false
+        type: string
```
Comment (marten-seemann) [inline .github/workflows/run-testplans.yml:17]: Similar to my comment on the go-libp2p PR, could this be merged into a single option if we make use of tags?

Comment (MarcoPolo) [reply .github/workflows/run-testplans.yml:17]: No. This was a misunderstanding.

**Comment by mxinden** [comment]
> Add emit only option, which always users to emit the compose spec that they can then manually run.

:rocket:

**Review by mxinden — APPROVED** [review]
Steps and code looks good to me. That said, I am not a GitHub action expert. I would assume Laurent and Piotr would be better as reviewers.

@jxs would you mind doing the changes of https://github.com/libp2p/go-libp2p/pull/1991/ for rust-libp2p?

@MarcoPolo I don't have an opinion on whether to block on https://github.com/libp2p/go-libp2p/pull/1991/ in rust-libp2p, or merge early here.

Great to see this happening. Thanks @MarcoPolo.

**Comment by MarcoPolo** [comment]
@galargh do you mind reviewing the GitHub workflow here?

**Review by galargh — COMMENTED** [review]
I reviewed `.github/workflows/run-testplans.yml` only. One note on `download-artifact` usage. Otherwise, looks good.

File: .github/workflows/run-testplans.yml [hunk 3]
```diff
@@ -18,15 +36,30 @@ jobs:
...
+      - uses: actions/download-artifact@v3
+        if: ${{ inputs.extra-versions != '' }}
+        with:
+          name: ${{ inputs.extra-versions }}
+          path: /tmp/extra-versions
+      - uses: actions/download-artifact@v3
+        if: ${{ inputs.image-tar != '' }}
+        with:
+          name: ${{ inputs.image-tar }}
+          path: /tmp/images/
```
[the remark this answers is not in the document]
Comment (MarcoPolo) [reply .github/workflows/run-testplans.yml:54]: Cool tip! Not needed for this PR, but I can imagine it being useful in other cases. I guess you have to keep in mind the 90 day limit of artifacts.

**Comment by thomaseizinger** [comment]
> @jxs would you mind doing the changes of [libp2p/go-libp2p#1991](https://github.com/libp2p/go-libp2p/pull/1991) for rust-libp2p?
> 
> @MarcoPolo I don't have an opinion on whether to block on [libp2p/go-libp2p#1991](https://github.com/libp2p/go-libp2p/pull/1991) in rust-libp2p, or merge early here.

Currently, we can't merge any PRs until this is fixed. See for example: https://github.com/libp2p/rust-libp2p/pull/3324

**Pull request merged**